packages/x/xgrammar/xmake.lua

```diff
@@ -0,0 +1,58 @@
+package("xgrammar")
+    set_homepage("https://xgrammar.mlc.ai/")
+    set_description("Fast, Flexible and Portable Structured Generation")
+    set_license("Apache-2.0")
+
+    add_urls("https://github.com/mlc-ai/xgrammar/archive/refs/tags/$(version).tar.gz",
+             "https://github.com/mlc-ai/xgrammar.git")
+    add_versions("v0.1.19", "f05f8d05b12b29523a2f299535a42180e665ce80109360a6afafc300d82f1b78")
+
+    add_configs("XGRAMMAR_BUILD_PYTHON_BINDINGS", {description = "Build Python bindings", default = false, type = "boolean"})
```

Comment:
```suggestion
    add_configs("shared", {description = "Build shared library.", default = false, type = "boolean", readonly = true})
    add_configs("XGRAMMAR_BUILD_PYTHON_BINDINGS", {description = "Build Python bindings", default = false, type = "boolean"})
```
In case your library is static only. See https://github.com/xmake-io/xmake-repo/pull/7169/files#r2095200249